osf/management/commands/sync_datacite_doi_metadata.py

```diff
@@ -10,31 +10,38 @@
 
 logger = logging.getLogger(__name__)
 
-def sync_datacite_doi_metadata(dry_run=True, retries=4):
+def sync_datacite_doi_metadata(dry_run=True, batch_size=100):
     content_type = ContentType.objects.get_for_model(Registration)
     reg_ids = Identifier.objects.filter(category='doi', content_type=content_type, deleted__isnull=True).values_list(
         'object_id',
         flat=True
     )
 
-    registrations = Registration.objects.exclude(id__in=reg_ids)
-    logger.info(f'{registrations.count()} registrations to mint')
+    registrations = Registration.objects.exclude(
+        deleted__isnull=False,
+        is_public=False,
+        moderation_state='withdrawn'
+    ).exclude(  # Strangely we are required to call exclude two times for a list comp.
+        id__in=reg_ids
+    )[:batch_size]
+    logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                f'{registrations.count()} registrations to mint')
+
     for registration in registrations:
-        if not dry_run:
+        retries = 4
```

Comment:
nit: this can be defined outside of the loop, since it's not being modified at all